<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8" />

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
<meta name="keywords" content="GrideaÈùôÊÄÅ‰∏™‰∫∫ÂçöÂÆ¢">
<meta name="description" content="Welcome to Qiaoji Xu&#39;s blog website">
<meta name="theme-color" content="#000">
<title>Qiaoji Xu - Lim</title>
<link rel="shortcut icon" href="/favicon.ico?v=1710985330249">
<link rel="stylesheet" href="/media/css/gemini.css">
<link rel="stylesheet" href="/media/fonts/font-awesome.css">
<link
  href="//fonts.googleapis.com/css?family=Monda:300,300italic,400,400italic,700,700italic|Roboto Slab:300,300italic,400,400italic,700,700italic|Rosario:300,300italic,400,400italic,700,700italic|PT Mono:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext"
  rel="stylesheet" type="text/css">

<link href="/media/hljs/styles/default.css"
  rel="stylesheet">

<link rel="stylesheet" href="/styles/main.css">

<script src="/media/hljs/highlight.js"></script>
<script src="https://cdn.jsdelivr.net/npm/velocity-animate@1.5.0/velocity.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/velocity-animate@1.5.0/velocity.ui.min.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css"
  integrity="sha384-zB1R0rpPzHqg7Kpt0Aljp8JPLqbXI3bhnPWROx27a9N0Ll6ZP/+DiW/UqRcLbRjq" crossorigin="anonymous">


<script defer src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js"
  integrity="sha384-y23I5Q6l+B6vatafAwxRu/0oK/79VlbSz7Q9aiSZUvyWYIYsd+qj+o24G5ZU2zJz" crossorigin="anonymous"></script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js"
  integrity="sha384-kWPLUVMOks5AQFrykwIup5lo0m3iMkkHrD0uJ4H5cjeGihAutqP0yW0J6dpFiVkI" crossorigin="anonymous"
  onload="renderMathInElement(document.body);"></script>





</head>

<body>
  <div class="head-top-line"></div>
  <div class="header-box">
    
<div class="gemini">
  <header class="header  ">
    <div class="blog-header box-shadow-wrapper bg-color " id="header">
      <div class="nav-toggle" id="nav_toggle">
        <div class="toggle-box">
          <div class="line line-top"></div>
          <div class="line line-center"></div>
          <div class="line line-bottom"></div>
        </div>
      </div>
      <div class="site-meta">       
        <div class="site-title">
          
            <a href="/" class="brand">
              <span>Qiaoji Xu - Lim</span>
            </a>  
          
        </div>
        
          <p class="subtitle">Lim</p>
        
      </div>
      <nav class="site-nav" id="site_nav">
        <ul id="nav_ul">
          
            
            
              
            
            <li class="nav-item ">
              
              
                <a href="/" target="_self">
                  <i class="fa fa-home"></i> About
                </a>
              
            </li>
          
            
            
              
            
            <li class="nav-item ">
              
              
                <a href="/Articles/" target="_self">
                  <i class="fa fa-globe"></i> Articles
                </a>
              
            </li>
          
            
            
              
            
            <li class="nav-item nav-item-active">
              
              
                <a href="/tags/" target="_self">
                  <i class="fa fa-tags"></i> Tag
                </a>
              
            </li>
          
            
            
              
            
            <li class="nav-item ">
              
              
                <a href="/archives/" target="_self">
                  <i class="fa fa-archive"></i> Category
                </a>
              
            </li>
          
          
            
              <li class="nav-item ">
                <a href="/friends/" target="_self">
                  
                    <i class="fa fa-address-book"></i> social links 
                  
                </a>
              </li>
            
          
          
            <li id="fa_search" class="nav-item">
              <a href="javascript:void(0);">
                <i class="fa fa-search"></i> <span class="language" data-lan="search">ÊêúÁ¥¢</span>
              </a>
            </li>
          
        </ul>
      </nav>
    </div>
  </header>
</div>

<script type="text/javascript"> 
 
  let showNav = true;

  let navToggle = document.querySelector('#nav_toggle'),
  siteNav = document.querySelector('#site_nav');
  
  function navClick() {
    let sideBar = document.querySelector('.sidebar');
    let navUl = document.querySelector('#nav_ul');
    navToggle.classList.toggle('nav-toggle-active');
    siteNav.classList.toggle('nav-menu-active');
    if (siteNav.classList.contains('nav-menu-active')) {
      siteNav.style = "height: " + (navUl.children.length * 42) +"px !important";
    } else {
      siteNav.style = "";
    }
  }

  navToggle.addEventListener('click',navClick);  
</script>
  </div>
  <div class="main-continer">
    
    <div
      class="section-layout  gemini">
      <div class="section-layout-wrapper">
        <div id="sidebarMeta" class="sidebar">
    
<div class="sidebar-wrapper box-shadow-wrapper bg-color">
  <div class="sidebar-item">
    <img class="site-author-image right-motion" src="/images/avatar.png"/>
    <p class="site-author-name">Qiaoji Xu - Lim</p>
    
    <div class="site-description right-motion">
      
      
      
        <p>&#34;La vie est simple, ne pas regretter apr√®s avoir prendu la d√©cision.&#34;</p>
      
      
    </div>
    
  </div>
  <div class="sidebar-item side-item-stat right-motion">
    <div class="sidebar-item-box">
      <a href="/archives/">
        
        <span class="site-item-stat-count">5</span>
        <span class="site-item-stat-name language" data-lan="article">ÊñáÁ´†</span>
      </a>
    </div>
    <div class="sidebar-item-box">
      <a href="">
        <span class="site-item-stat-count">3</span>
        <span class="site-item-stat-name language" data-lan="category">ÂàÜÁ±ª</span>
      </a>
    </div>
    <div class="sidebar-item-box">
      <a href="/tags/">
        <span class="site-item-stat-count">3</span>
        <span class="site-item-stat-name language" data-lan="tag">Ê†áÁ≠æ</span>
      </a>
    </div>
  </div>
  
    
  
  
    <div class="sidebar-item sidebar-item-social">
      <div class="social-item">
        
          
            <a href="https://github.com/Qiaojilim">
              <i class="fa fa-github-alt" title="github"></i>
            </a>
          
            <a href="https://www.linkedin.com/in/qiaoji-lim-xu-53156b245/">
              <i class="fa fa-linkedin" title="LinkedIn"></i>
            </a>
          
            <a href="limqiaojixu@gmail.com">
              <i class="fa envelope" title="Email"></i>
            </a>
          
        
        
      </div>
    </div>
  



  <div style="width: 100%; position: relative;">
    <canvas id="canvasDiyBlock" style="width:100%;">ÂΩìÂâçÊµèËßàÂô®‰∏çÊîØÊåÅcanvasÔºåËØ∑Êõ¥Êç¢ÊµèËßàÂô®ÂêéÂÜçËØï</canvas>
    <script src="/media/js/magic/clock.js"></script>
  </div>

</div>
</div>
<script>
  let sidebarMeta = document.querySelector('#sidebarMeta');
  let scheme = 'gemini';
  let sidebarWrapper = document.querySelector('.sidebar-wrapper');
  if (sidebarMeta && (scheme === 'pisces' || scheme === 'gemini')) {
    document.addEventListener('scroll', function(e) {
      if (document.scrollingElement.scrollTop > parseInt(sidebarMeta.style.marginTop) + 10) {
        sidebarWrapper.classList.add('home-sidebar-fixed')
      } else {
        sidebarWrapper.classList.remove('home-sidebar-fixed')
      }
    });
  }
  </script>
        <div class="section-box box-shadow-wrapper">
          <section class="section tags-section bg-color posts-expand">
            <div class="tags-box">
  <div class="tag-cloud-title language" data-lan="cloud-tags" data-count="3">
    ÁõÆÂâçÂÖ±ËÆ° 3 ‰∏™Ê†áÁ≠æ
  </div>
  <div class="tag-cloud-list" id="tags">
    
      
      
        
      
        
          
            
      
        
          
        
          
            
      
        
      
        
          
        
          
            
      
        
          
        
          
            
      
      <a class="cloud-tag" href="https://qiaojilim.github.io/tag/eB4F0sw8A/" rel="4">code notes</a>
    
      
      
        
      
        
          
        
      
        
          
            
      
        
      
        
          
        
          
        
      
        
          
        
          
        
      
      <a class="cloud-tag" href="https://qiaojilim.github.io/tag/FC3t5kgaD/" rel="1">Xcode</a>
    
      
      
        
      
        
          
        
      
        
          
        
          
        
      
        
      
        
          
            
      
        
          
            
      
      <a class="cloud-tag" href="https://qiaojilim.github.io/tag/f2Cgtx5tT/" rel="2">circos</a>
    
  </div>
</div>
<script>

  let fontStart = 12, fontEnd = 30, colorStart = '#666', colorEnd = '#000';
  let tags = document.querySelector('#tags').children;
  let counts = [];
  for(let i = 0; i < tags.length; i++) {
    counts.push(tags[i].rel);
  }
  let lowest = Math.min.apply(null,counts), highest = Math.max.apply(null,counts);;
  let range = highest - lowest;
  if (range === 0) {
    range = 1;
  }
  let fontIncr, colorIncr;
  fontIncr = (fontEnd - fontStart) / range;
  
  let sColorArr = toRGB(colorStart), eColorArr = toRGB(colorEnd);

  for(let i = 0; i < tags.length; i++) {
    let disCount = tags[i].rel - lowest;
    let fontSize = fontStart + disCount * fontIncr;
    let style = 'font-size: ' + fontSize + 'px;';
    let arr = [];
    for(let c = 0; c < sColorArr.length; c++) {
      if (disCount === 0) {
        disCount = 1;
      }
      let val = (sColorArr[c] - eColorArr[c]) / range * disCount;
      if (val >= 100) {
        val = eColorArr[c];
      } else if (val <= 0) {
        val = sColorArr[c];
      }
      arr[c] = Math.floor(val);
    }
    style += 'color: ' + toHex(arr);
    tags[i].style = style;
  }

  function toRGB(code) {
    if (code.length === 4) {
      code = code.replace(/(\w)(\w)(\w)/gi, "\$1\$1\$2\$2\$3\$3");
    }
    var hex = /(\w{2})(\w{2})(\w{2})/.exec(code);
    return [parseInt(hex[1], 16), parseInt(hex[2], 16), parseInt(hex[3], 16)];
  }

  function toHex(ary) {
    return "#" + ((1 << 24) + (ary[0] << 16) + (ary[1] << 8) + ary[2]).toString(16).slice(1);
  }

</script>
          </section>
        </div>
      </div>
    </div>
    <div class="footer-box">
  <footer class="footer">
    <div class="copyright">
      Powered by <a href="https://github.com/getgridea/gridea" target="_blank">Gridea</a> | ¬© 2019-2020 Theme By <a
        href="https://github.com/hsxyhao/gridea-theme-next" target="_blank">HsxyHao</a>
    </div>
    <div class="poweredby">
      Powered by <a href="https://github.com/getgridea/gridea" target="_blank">Gridea</a>
    </div>
  </footer>
  
  
  <div class="gemini back-to-top" id="back_to_top">
    <i class="fa fa-arrow-up"></i>
    
    <span class="scrollpercent">
      <span id="back_to_top_text">0</span>%
    </span>
    
  </div>
  
  
  
</div>
<script>

  let sideBarOpen = 'sidebar-open';
  let body = document.body;
  let back2Top = document.querySelector('#back_to_top'),
    back2TopText = document.querySelector('#back_to_top_text'),
    drawerBox = document.querySelector('#drawer_box'),
    rightSideBar = document.querySelector('.sidebar'),
    viewport = document.querySelector('body');

  function scrollAnimation(currentY, targetY) {

    let needScrollTop = targetY - currentY
    let _currentY = currentY
    setTimeout(() => {
      const dist = Math.ceil(needScrollTop / 10)
      _currentY += dist
      window.scrollTo(_currentY, currentY)
      if (needScrollTop > 10 || needScrollTop < -10) {
        scrollAnimation(_currentY, targetY)
      } else {
        window.scrollTo(_currentY, targetY)
      }
    }, 1)
  }

  back2Top.addEventListener("click", function (e) {
    scrollAnimation(document.scrollingElement.scrollTop, 0);
    e.stopPropagation();
    return false;
  });

  window.addEventListener('scroll', function (e) {
    let percent = document.scrollingElement.scrollTop / (document.scrollingElement.scrollHeight - document.scrollingElement.clientHeight) * 100;
    if (percent > 1 && !back2Top.classList.contains('back-top-active')) {
      back2Top.classList.add('back-top-active');
    }
    if (percent == 0) {
      back2Top.classList.remove('back-top-active');
    }
    if (back2TopText) {
      back2TopText.textContent = Math.floor(percent);
    }
  });


  let hasCacu = false;
  window.onresize = function () {
    calcuHeight();
  }

  function calcuHeight() {
    // Âä®ÊÄÅË∞ÉÊï¥Á´ôÁÇπÊ¶ÇËßàÈ´òÂ∫¶
    if (!hasCacu && back2Top.classList.contains('pisces') || back2Top.classList.contains('gemini')) {
      let sideBar = document.querySelector('.sidebar');
      let navUl = document.querySelector('#site_nav');
      sideBar.style = 'margin-top:' + (navUl.offsetHeight + navUl.offsetTop + 15) + 'px;';
      hasCacu = true;
    }
  }
  calcuHeight();

  let open = false, MOTION_TIME = 300, RIGHT_MOVE_DIS = '320px';

  if (drawerBox) {
    let rightMotions = document.querySelectorAll('.right-motion');
    let right = drawerBox.classList.contains('right');

    let transitionDir = right ? "transition.slideRightIn" : "transition.slideLeftIn";

    let openProp, closeProp;
    if (right) {
      openProp = {
        paddingRight: RIGHT_MOVE_DIS
      };
      closeProp = {
        paddingRight: '0px'
      };
    } else {
      openProp = {
        paddingLeft: RIGHT_MOVE_DIS
      };
      closeProp = {
        paddingLeft: '0px'
      };
    }

    drawerBox.onclick = function () {
      open = !open;
      window.Velocity(rightSideBar, 'stop');
      window.Velocity(viewport, 'stop');
      window.Velocity(rightMotions, 'stop');
      if (open) {
        window.Velocity(rightSideBar, {
          width: RIGHT_MOVE_DIS
        }, {
          duration: MOTION_TIME,
          begin: function () {
            window.Velocity(rightMotions, transitionDir, {});
          }
        })
        window.Velocity(viewport, openProp, {
          duration: MOTION_TIME
        });
      } else {
        window.Velocity(rightSideBar, {
          width: '0px'
        }, {
          duration: MOTION_TIME,
          begin: function () {
            window.Velocity(rightMotions, {
              opacity: 0
            });
          }
        })
        window.Velocity(viewport, closeProp, {
          duration: MOTION_TIME
        });
      }
      for (let i = 0; i < drawerBox.children.length; i++) {
        drawerBox.children[i].classList.toggle('muse-line');
      }
      drawerBox.classList.toggle(sideBarOpen);
    }
  }

  // ÈìæÊé•Ë∑≥ËΩ¨
  let newWindow = 'false'
  if (newWindow === 'true') {
    let links = document.querySelectorAll('.post-body a')
    links.forEach(item => {
      if (!item.classList.contains('btn')) {
        item.setAttribute("target", "_blank");
      }
    })
  }

  let faSearch = document.querySelector('#fa_search');
  faSearch.addEventListener('click', function () {
    document.querySelector('#search_mask').style = ''
  })

  // ‰ª£Á†ÅÈ´ò‰∫Æ
  hljs.initHighlightingOnLoad();
  
  // Á¶ªÂºÄÂΩìÂâçÈ°µtitleÂèòÂåñ
  var leaveTitle = "";
  if (leaveTitle) {
    document.addEventListener('visibilitychange', function () {
      if (document.visibilityState == 'hidden') {
        normal_title = document.title;
        document.title = leaveTitle;
      } else {
        document.title = normal_title;
      }
    });
  }

</script>
  </div>
</body>

<div class="search-mask" id="search_mask" style="display: none;">
  <div class="search-box">
    <div class="search-title">
      <i class="fa fa-search"></i>
      <div class="input-box">
        <input id="search" type="text" class="language" data-lan="search" placeholder="ÊêúÁ¥¢">
      </div>
      <i id="close" class="fa fa-times-circle"></i>
    </div>
    <div class="stat-box">
      <span id="stat_count">0</span><span class="language" data-lan="stat">Êù°Áõ∏ÂÖ≥Êù°ÁõÆÔºå‰ΩøÁî®‰∫Ü</span><span id="stat_times">0</span><span class="language" data-lan="stat-time">ÊØ´Áßí</span>
      <hr>
    </div>
    <div class="result" id="result">
      
      <div class="item">
        <a class="result-title" style="opacity: 0;" href="https://qiaojilim.github.io/post/git-to-publish-a-project-and-update-later/"" data-c="
          &lt;h2 id=&#34;creating-a-new-repository-on-your-github-account-without-any-file&#34;&gt;* Creating a new repository on your github account without any file&lt;/h2&gt;
&lt;ol&gt;
&lt;li&gt;logging into your github account page the click &amp;quot;+&amp;quot; at the right upper corner:&lt;br&gt;
&lt;img src=&#34;https://qiaojilim.github.io/post-images/1619111991415.png&#34; alt=&#34;&#34; loading=&#34;lazy&#34;&gt;&lt;/li&gt;
&lt;li&gt;Then typing repository name and creat it without any files in it.&lt;/li&gt;
&lt;/ol&gt;
&lt;h2 id=&#34;push-the-project-to-the-created-repository&#34;&gt;* Push the project to the created repository&lt;/h2&gt;
&lt;ol&gt;
&lt;li&gt;Pushing all files from local dir to github&lt;/li&gt;
&lt;/ol&gt;
&lt;pre&gt;&lt;code&gt;cd /your/file/dir
git init
git add .
git commit -m &amp;quot;first update&amp;quot;
#copy http address from the repo
git remote add origin https://github.com/Qiaojilim/test.git 
git branch -M main
git push -u origin main
&lt;/code&gt;&lt;/pre&gt;
&lt;ol start=&#34;2&#34;&gt;
&lt;li&gt;Merging all files from the repo to local dir and upload them back to old repo&lt;/li&gt;
&lt;/ol&gt;
&lt;pre&gt;&lt;code&gt;#if we need to create a new folder from github then we use git clone https://github.com/Qiaojilim/RACCROCHE-update.git
cd /your/file/dir
#pulling all files from existing repo on github
git remote add upstream https://github.com/Qiaojilim/RACCROCHE-update.git
git pull upstream main
#adding all files from local dir to github repo
git add .
git commit -m &amp;quot;my commitment&amp;quot;
git push origin main
&lt;/code&gt;&lt;/pre&gt;
">Git to publish a project and update from local</a>
      </div>
      
      <div class="item">
        <a class="result-title" style="opacity: 0;" href="https://qiaojilim.github.io/post/publish-and-packaging-python-project/"" data-c="
          &lt;p&gt;If we want to publish a python software and let other people to use our tool as well, then we need to package our code files and than publish it.&lt;/p&gt;
&lt;h2 id=&#34;layout-of-package&#34;&gt;Layout of package&lt;/h2&gt;
&lt;p&gt;packaging_tutorial&lt;br&gt;
|‚Äî‚Äî‚Äî‚Äî LICENSE&lt;br&gt;
|‚Äî‚Äî‚Äî‚Äî README&lt;br&gt;
|‚Äî‚Äî‚Äî‚Äî example_pkg&lt;br&gt;
|     |________   _&lt;em&gt;init _&lt;/em&gt;.py&lt;br&gt;
|‚Äî‚Äî‚Äî‚Äî setup.py&lt;br&gt;
|‚Äî‚Äî‚Äî‚Äî test&lt;br&gt;
Once we create this structure, we need to &lt;code&gt;cd packaging_tutorial&lt;/code&gt;&lt;/p&gt;
&lt;ul&gt;
&lt;li&gt;Project structure needs to be in place. Usually, the root of the repository contains a folder with the name of the repository ‚Äî this is where the core project code should be. What‚Äôs outside of this folder is the extra code that‚Äôs necessary to run and build the package (tests, documentation, etc.)&lt;/li&gt;
&lt;li&gt;The core folder should include one (or more) module(s) and an _&lt;em&gt;init&lt;/em&gt;_.py file which imports the classes/functions you‚Äôll want the end user to have access to. This file can also include the version of the package, to make it easily accessible to the end user.&lt;/li&gt;
&lt;li&gt;example_pkg :  _&lt;em&gt;init _&lt;/em&gt;.py  files are required to make Python treat directories containing the file as packages. _&lt;em&gt;init&lt;/em&gt;_.py can just be an empty file, more info please check :  &lt;a href=&#34;https://docs.python.org/3/tutorial/modules.html#packages&#34;&gt;python packeges and modules&lt;/a&gt;&lt;/li&gt;
&lt;li&gt;test folder : containg all unit test files, we can leav it empty now.&lt;/li&gt;
&lt;/ul&gt;
&lt;h3 id=&#34;choose-license&#34;&gt;Choose license&lt;/h3&gt;
&lt;p&gt;The usual licenses for open sources projects are GNU, MIT, BSD, detalied see &lt;a href=&#34;https://choosealicense.com/&#34;&gt;License&lt;/a&gt;&lt;/p&gt;
&lt;h3 id=&#34;readme-file&#34;&gt;README file&lt;/h3&gt;
&lt;p&gt;A README is a txt file that contains informations which is commonly required to understand what the project is about.  The most common format is &lt;a href=&#34;https://en.wikipedia.org/wiki/Markdown&#34;&gt;Markdown&lt;/a&gt;&lt;br&gt;
I usually use online readme.md editor &lt;a href=&#34;https://dillinger.io&#34;&gt;Dillinger&lt;/a&gt;&lt;br&gt;
Instruction to write a beautiful README.md please check my another article and here is an example of my readme.md&lt;br&gt;
&lt;a href=&#34;https://github.com/Qiaojilim/raccroche_demo&#34;&gt;Example of README.md&lt;/a&gt;&lt;/p&gt;
&lt;h3 id=&#34;create-a-setuppy-file&#34;&gt;Create a Setup.py file&lt;/h3&gt;
&lt;pre&gt;&lt;code&gt;from setuptools import setup
def parse_requirements(requirements):
    &amp;quot;&amp;quot;&amp;quot;
    Returns the list of requirements found in the requirements file
    &amp;quot;&amp;quot;&amp;quot;
    with open(requirements, &amp;quot;r&amp;quot;) as req_file:
        return [l.strip(&#39;\n&#39;) for l in req_file if l.strip(&#39;\n&#39;)
                and not l.startswith(&#39;#&#39;)]
#pacages: each folder that with a __init__.py file in its dir, here contigs_alg is a #subofolder under raccroche with __init__.py
packages = [&#39;raccroche&#39;, &#39;raccroche.contigs_alg&#39;] 
requires = parse_requirements(&amp;quot;requirements.txt&amp;quot;)
#scripts 
scripts = [&#39;bin/raccroche_main&#39;] 

classifiers = [
    &amp;quot;Environment :: Console&amp;quot;,
    &amp;quot;Intended Audience :: Science/Research&amp;quot;,
    &amp;quot;License :: OSI Approved :: BSD License&amp;quot;,
    &amp;quot;Programming Language :: Python :: 3&amp;quot;,
    &amp;quot;Operating System :: OS Independent&amp;quot;,
    &amp;quot;Topic :: Scientific/Engineering :: Bio-Informatics&amp;quot;,
]

with open(&#39;README.md&#39;) as f:
    long_description = f.read()

setup(
    name=&#39;raccroche&#39;,
    packages=packages,
    version=&#39;1.0.0&#39;,
    description=&#39;raccroche demo: Reconstruction of AnCestral COntigs and CHromosomEs&#39;,
    long_description=long_description,
    author=&#39;Qiaoji Xu, Lingling Jin&#39;,
    author_email=&#39;qxu062@uottawa.ca, lingling.jin@cs.usask.ca&#39;,
    license=&#39;BSD-2-Clause&#39;,
    platforms=&#39;OS Independent&#39;,
    package_data={&#39;&#39;: [&#39;LICENSE&#39;]},
    download_url=&#39;https://github.com/Qiaojilim/raccroche_demo&#39;,
    url=&#39;https://github.com/Qiaojilim/raccroche_demo&#39;,
    scripts=scripts,
    include_package_data=True,
    install_requires=requires,
    #tests_require=[&#39;pytest&#39;],
    #cmdclass={&#39;tests&#39;: PyTest},
    classifiers=classifiers,
    python_requires=&#39;&amp;gt;=3&#39;
)
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;&lt;strong&gt;Notes&lt;/strong&gt;:&lt;br&gt;
If you want metadata (from the repository) to be downloaded when anyone installs the package, you should add these through the package_data argument.&lt;/p&gt;
&lt;h3 id=&#34;generating-distribution-archives&#34;&gt;Generating distribution archives&lt;/h3&gt;
&lt;p&gt;The next step is to generate distribution packages for the package. These are archives that are uploaded to the Package Index and can be installed by pip.&lt;/p&gt;
&lt;p&gt;Make sure you have the latest versions of setuptools and wheel installed:&lt;/p&gt;
&lt;p&gt;&lt;code&gt;python3 -m pip install --user --upgrade setuptools wheel&lt;/code&gt;&lt;/p&gt;
&lt;p&gt;Now run this command from the same directory where setup.py is located:&lt;br&gt;
&lt;code&gt;python3 setup.py sdist bdist_wheel&lt;/code&gt;&lt;/p&gt;
&lt;p&gt;This command should output a lot of text and once completed should generate two files in the dist directory:&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;dist/
  example_pkg_YOUR_USERNAME_HERE-0.0.1-py3-none-any.whl
  example_pkg_YOUR_USERNAME_HERE-0.0.1.tar.gz
&lt;/code&gt;&lt;/pre&gt;
&lt;h3 id=&#34;local-installation&#34;&gt;Local installation&lt;/h3&gt;
&lt;p&gt;In dist folder, decompress the zip &lt;code&gt;example_pkg_YOUR_USERNAME_HERE-0.0.1.tar.gz&lt;/code&gt;&lt;br&gt;
and &lt;code&gt;cd /this /decompressed /folder /where /the /setup.py /locate&lt;/code&gt;&lt;br&gt;
run: &lt;code&gt;sudo python3 setup.py install&lt;/code&gt;&lt;br&gt;
We install it in our local dir &lt;code&gt;/Library/Python/3.8/site-packages&lt;/code&gt;&lt;/p&gt;
&lt;h3 id=&#34;uninstall&#34;&gt;Uninstall&lt;/h3&gt;
&lt;p&gt;&lt;code&gt;sudo pip3 uninstall program_name&lt;/code&gt;&lt;/p&gt;
&lt;p&gt;References links:&lt;br&gt;
https://medium.com/free-code-camp/from-a-python-project-to-an-open-source-package-an-a-to-z-guide-c34cb7139a22&lt;br&gt;
https://www.jianshu.com/p/b19330b9ab14&lt;/p&gt;
">Publish and Packaging Python project - 1</a>
      </div>
      
      <div class="item">
        <a class="result-title" style="opacity: 0;" href="https://qiaojilim.github.io/post/how-to-access-github-with-xcode11-and-upload-projects/"" data-c="
          &lt;p&gt;Let&#39;s go over how to upload projects on github by Xcode11 step by step.&lt;/p&gt;
&lt;h2 id=&#34;creating-a-new-repository-under-your-github-account&#34;&gt;creating a new repository under your github account:&lt;/h2&gt;
&lt;ol&gt;
&lt;li&gt;clicking &#39;+&#39; button and &amp;quot;New repository&amp;quot; in the fowllowing picture then it will pop up a window to fill in information about the new repository.&lt;/li&gt;
&lt;/ol&gt;
&lt;p&gt;&lt;img src=&#34;https://qiaojilim.github.io/post-images/1603588888242.png&#34; alt=&#34;&#34; loading=&#34;lazy&#34;&gt;&lt;br&gt;
2. getting your token code from you github account by clicking &#39;setting&#39;, than on the leftside, find and click &#39;Developer setting&#39;. Finally, you can see &#39;Personal access tokens&#39; where we can copy our tokens as passward afterwards.&lt;br&gt;
&lt;img src=&#34;https://qiaojilim.github.io/post-images/1603589053807.png&#34; alt=&#34;&#34; loading=&#34;lazy&#34;&gt;&lt;/p&gt;
&lt;h2 id=&#34;using-xcode-to-set-up&#34;&gt;using Xcode to set up&lt;/h2&gt;
&lt;p&gt;If you do not have Xcode, you can donwload it  either from apple store or this link: https://developer.apple.com/download/more/.&lt;/p&gt;
&lt;ol&gt;
&lt;li&gt;open Xcode app, on the left coner, under &#39;Xcode&#39; clicking preference&lt;br&gt;
&lt;img src=&#34;https://qiaojilim.github.io/post-images/1603589952059.png&#34; alt=&#34;&#34; loading=&#34;lazy&#34;&gt;&lt;br&gt;
On the pop up window, click &#39;Accounts@&#39; upper left corner, and choose &#39;+&#39; to add your github account. Next, you will need to input your github username and token copied from the first part. after done, you will see this:&lt;br&gt;
&lt;img src=&#34;https://qiaojilim.github.io/post-images/1603590524978.png&#34; alt=&#34;&#34; loading=&#34;lazy&#34;&gt;&lt;/li&gt;
&lt;/ol&gt;
&lt;ul&gt;
&lt;li&gt;connecting your xcode and the specified github repostority&lt;br&gt;
under &amp;quot;Xcode-Source Control --&amp;gt;Clone&amp;quot;Ôºåall repositories show up nect jsut choose which one you want to connect to colone.&lt;br&gt;
open your project in xcode and unclick &#39;create Git respository on my mac&#39;, move your project fold into the cloned folder and choose:&lt;br&gt;
&amp;quot;Source control  &amp;gt;&amp;gt; Commit &amp;gt;&amp;gt; Push&amp;quot;&lt;br&gt;
Then go back to your github web, you are gonna see all updated files.&lt;/li&gt;
&lt;/ul&gt;
&lt;p&gt;&lt;strong&gt;Reference link:&lt;/strong&gt;&lt;br&gt;
http://www.cnblogs.com/mukekeheart/p/8309578.html&lt;/p&gt;
"> Access Github with Xcode11 and upload projects</a>
      </div>
      
      <div class="item">
        <a class="result-title" style="opacity: 0;" href="https://qiaojilim.github.io/post/about/"" data-c="
          &lt;p&gt;&lt;strong&gt;Qiaoji Xu&lt;/strong&gt;&lt;br&gt;
üè† Toronto, Cananda&lt;br&gt;
üì¨ limqiaojixu@gmail.com&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Profile&lt;/strong&gt;&lt;br&gt;
As a forward-thinking AI/Data Science Consultant and Entrepreneur, I leverage a robust Ph.D. background in Mathematics and Statistics with over 5 years of multifaceted experience across finance and engineering sectors. My expertise encompasses advanced module and pipeline development, statistical analysis, risk management, and algorithmic optimization, underscored by profound proficiency in Python, R, and Machine Learning technologies for large datasets. My entrepreneurial spirit is driven by a passion for devising innovative algorithms and predictive models that not only solve complex problems but also enhance business efficiency and strategic goals. I thrive in environments that challenge the status quo, inspiring teams towards the creation of groundbreaking solutions in comparative genomics, finance, and beyond.&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;EDUCATION&lt;/strong&gt;&lt;br&gt;
2019 - 2023&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;PhD, Science, Statistics, University of Ottawa, Canada        &lt;br&gt;&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Professional Experience&lt;/strong&gt;&lt;br&gt;
2024-01 - present&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt; Manager, Data Scientist	 @Scotiabank, Canada&lt;/p&gt;
&lt;ul&gt;
&lt;li&gt;Conduct model impact assessment in platform conversion based on product portfolios through PySpark by developing statistical solutions and recommend improvements to estimation techniques and/or forecasting models by python and SQL for platform conversion.&lt;/li&gt;
&lt;li&gt;Collaborate with risk management, vendors, technology, foreign counterparts and compliance audit to ensure effective governance, measurement &amp;amp; communication of model performance.&lt;/li&gt;
&lt;li&gt;Documented improvements for model development and monitoring for review and validation.&lt;/li&gt;
&lt;/ul&gt;
&lt;p&gt;2023&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt; Big Data Modelling Analyst      @Ciena, Canada&lt;/p&gt;
&lt;ul&gt;
&lt;li&gt;Led a project focused on analyzing large network datasets, utilizing various statistical modeling techniques and algorithm optimization to maximize network resiliency, predict outage occurrences and minimizes failure risks.&lt;/li&gt;
&lt;li&gt;Collaborated closely with the product team to develop a Python-based production simulator embedded into the app, resulting in improvement in reasonability and user experience.&lt;/li&gt;
&lt;li&gt;Developed a machine learning model to assess the likelihood of fiber cuts on a geographic map with 95%+ accuracy, embedding new functions for customers likely to attrit by servicing relevant product features.&lt;/li&gt;
&lt;li&gt;Conducted in-depth analysis of time series big datasets using AWS database and SQL, improved outage prediction accuracy by 20%.&lt;/li&gt;
&lt;li&gt;Effectively communicated with PMs to cooperate three data engineers in project planning, development and execution, ensuring smooth and efficient project delivery.&lt;/li&gt;
&lt;/ul&gt;
&lt;p&gt;&lt;strong&gt;Research Experience&lt;/strong&gt;&lt;br&gt;
2019 - 2023&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Research Scientist, Sankoff lab at University of Ottawa  &lt;br&gt;  &lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Collaborate with global labs for computainal genomics projects with publication on top tier journals.&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Constructe aumative pipeline of exploring an efficient method to construct ancestral plant chromosomes and gene orders with longer contigs and higher coverage on genome about graph theory model, unsupervised learning and algoritm optimization.&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Executed computationally intensive ML algorithms on Google Cloud Platform and utilize multiprocessing and parallel programming in python for optimizing personalized modules based on 1 TB genome data.&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Teaching Experience&lt;/strong&gt;&lt;br&gt;
2019 - 2022&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Teaching Assistant, Demonstrator, University of Ottawa&lt;br&gt;
2020/01 - 2020/04     Math Help Center, University of Ottawa&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Academic and Honours&lt;/strong&gt;&lt;br&gt;
2019 - 2022&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;International Doctoral Scholarship&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Courses Average Grade above A&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Publications&lt;/strong&gt;			&lt;br&gt;
Xu,Q.;Jin,L.;Zheng,C.;Leebens-Mack,J.H.;Sankoff,D.RACCROCHE: Ancestral flowering plant chromosomes and gene ordersbased on generalized adjacencies and chromosomal gene co-occurrences.  Proceedings of the 10th International Conference on Computational Advances in Bio and Medical Sciences. InLecture Notes in Bioinformatics; Jha, S.K., Mandoiu, I., Rajasekaran, S.,Sahni, S., Skums, P., Zelikovsky, A., Eds.;2021; Volume 12686.&lt;/p&gt;
&lt;p&gt;&lt;a href=&#34;https://www.mdpi.com/1999-4893/14/6/160/htm&#34;&gt;Validation of automated chromosome recovery in the reconstruction of ancestral gene order (Q. Xu, L. Jin, J.H. Leeben-Mack &amp;amp; D. Sankoff)&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;&lt;a href=&#34;https://doi.org/10.1038/s41467-022-28312-w&#34;&gt;Chanderbali, A.; et al.  Buxus and Tetracentron genomes help resolve eudicot genome history. Nat Commun 13, 643 (2022)&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Software&lt;/strong&gt;&lt;br&gt;
&lt;a href=&#34;https://github.com/jin-repo/RACCROCHE&#34;&gt;RACCROCHE: Reconstruction of AnCestral COntigs and CHromosomEs&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Conferences Attended&lt;/strong&gt;&lt;br&gt;
2023&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;RECOMCG, Istanbul, Turkey&lt;br&gt;
2022&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Marine Biological Laboratory, US&lt;br&gt;
2020&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;ISMB Conference, virtual&lt;br&gt;
2019&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;RECOMCG, Montpellier, France&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Languages&lt;/strong&gt; &lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Fluent in English and Chinese (spoken and written)&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mtext&gt;¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†¬†&lt;/mtext&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;~~~~~~~~~~~~~~~~~~~~~~~~~~&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;span class=&#34;mspace nobreak&#34;&gt;¬†&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;Beginner in French&lt;/p&gt;
">CV</a>
      </div>
      
      <div class="item">
        <a class="result-title" style="opacity: 0;" href="https://qiaojilim.github.io/post/how-to-install-circos-on-mac-os-system/"" data-c="
          &lt;h1 id=&#34;steps-to-install-circos-in-mac-os&#34;&gt;Steps to install Circos in Mac OS&lt;/h1&gt;
&lt;ul&gt;
&lt;li&gt;First, from oficial website: https://circos.ca dowload the lastest version of CIRCOS for Mac. And install it in a folder ~/app, if the path of that folder is not in the PATH  of mac then we need to add it to by the instructions of the link below&lt;br&gt;
https://www.architectryan.com/2012/10/02/add-to-the-path-on-mac-os-x-mountain-lion/&lt;/li&gt;
&lt;li&gt;We can use brew install libpng, libjpeg, freetype, libgd.&lt;/li&gt;
&lt;/ul&gt;
&lt;pre&gt;&lt;code&gt;    brew install libpng
    brew install libjpeg
    brew install freetype
    brew install libgd
&lt;/code&gt;&lt;/pre&gt;
&lt;ul&gt;
&lt;li&gt;Test and install missing parts of perl moudles&lt;br&gt;
&lt;code&gt;circos -modules&lt;/code&gt;&lt;br&gt;
&amp;quot; missing            Font::TTF::Font &amp;quot;&lt;br&gt;
This mean we do not have Font::TTF::Font  module so we need to install all missing modules by:&lt;/li&gt;
&lt;/ul&gt;
&lt;pre&gt;&lt;code&gt;    sudo cpan
    cpan[1]&amp;gt; install Config::General
    cpan[2]&amp;gt; install Font::TTF::Font
    ...
    cpan[*]&amp;gt; exit 
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;But we may missing module GD and GD::Polyline so that we can install it manaually by:&lt;br&gt;
wget http://search.cpan.org/CPAN/authors/id/L/LD/LDS/GD-2.51.tar.gz&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;   tar xfvz ./GD-2.51.tar.gz
   cd GD-2.51/
   perl Makefile.PL    
   make
   sudo make install
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;In the end we test the sample of circos by:&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;   cd ~/Software/circos/current/example
   ./run
   open circos.png
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;If we get a nice pic then this means we install it successfully!&lt;/p&gt;
&lt;figure data-type=&#34;image&#34; tabindex=&#34;1&#34;&gt;&lt;img src=&#34;https://qiaojilim.github.io/post-images/1602883560231.png&#34; alt=&#34;&#34; loading=&#34;lazy&#34;&gt;&lt;/figure&gt;
&lt;p&gt;reference link: https://www.jianshu.com/p/d1715b25f9d1&lt;br&gt;
https://kylase.github.io/CircosAPI/os-x-installation-guide/&lt;/p&gt;
">How to install Circos on Mac OS system</a>
      </div>
      
      <div class="item">
        <a class="result-title" style="opacity: 0;" href="https://qiaojilim.github.io/post/How to plot synteny blocks by CIRCOS/"" data-c="
          &lt;h2 id=&#34;creating-multiple-plots-for-one-genome-such-as-heatmap-histgram-line-map-and-so-on&#34;&gt;Creating multiple plots for one genome such as heatmap, histgram, line map and so on.&lt;/h2&gt;
&lt;p&gt;-Input files:&lt;br&gt;
Collecting all file in the current dir: ~/app/circos&lt;br&gt;
i): karyotype.genome.txt  format is : &amp;quot;chr - ID LABEL START END COLOR&amp;quot;&lt;br&gt;
&lt;code&gt;chr - aco1 1 0 37743429 rdylbu-11-div-1&lt;/code&gt;&lt;br&gt;
Here the color of chromosome at the last col of karyotype.genomeA.txt file, we can serch color on the web: http://colorbrewer2.org/&lt;br&gt;
The format of color is: &lt;code&gt;Palette-Numcolors-Type-Idx&lt;/code&gt;:&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;P&lt;/mi&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mi&gt;e&lt;/mi&gt;&lt;mi&gt;t&lt;/mi&gt;&lt;mi&gt;t&lt;/mi&gt;&lt;mi&gt;e&lt;/mi&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;Palette&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0.69444em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.13889em;&#34;&gt;P&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;a&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.01968em;&#34;&gt;l&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;e&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;t&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;t&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;e&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;: name of color system like : gnbu&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;N&lt;/mi&gt;&lt;mi&gt;u&lt;/mi&gt;&lt;mi&gt;m&lt;/mi&gt;&lt;mi&gt;c&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mi&gt;s&lt;/mi&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;Numcolors&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0.69444em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.10903em;&#34;&gt;N&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;u&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;m&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;c&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;o&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.01968em;&#34;&gt;l&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;o&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.02778em;&#34;&gt;r&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;s&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;: number of data classes, eg: 3,4,5,6,11&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;T&lt;/mi&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;mi&gt;p&lt;/mi&gt;&lt;mi&gt;e&lt;/mi&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;Type&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0.8777699999999999em;vertical-align:-0.19444em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.13889em;&#34;&gt;T&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.03588em;&#34;&gt;y&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;p&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;e&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;: = nature of your data : div~(diverging), seq~(sequential), qual~(qualitative)&lt;br&gt;
&lt;span class=&#34;katex&#34;&gt;&lt;span class=&#34;katex-mathml&#34;&gt;&lt;math&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;I&lt;/mi&gt;&lt;mi&gt;d&lt;/mi&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;/mrow&gt;&lt;annotation encoding=&#34;application/x-tex&#34;&gt;Idx&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class=&#34;katex-html&#34; aria-hidden=&#34;true&#34;&gt;&lt;span class=&#34;base&#34;&gt;&lt;span class=&#34;strut&#34; style=&#34;height:0.69444em;vertical-align:0em;&#34;&gt;&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34; style=&#34;margin-right:0.07847em;&#34;&gt;I&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;d&lt;/span&gt;&lt;span class=&#34;mord mathdefault&#34;&gt;x&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;: cchoosing which index of color to out of Numcolors.&lt;/p&gt;
&lt;p&gt;ii): creating etc foldr under  ~/app/circos to contain all other configures files, like links.conf and ticks.con&lt;br&gt;
iii):  creating ticks.conf in etc folder to input all parameters for plotting ticks as:&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;# this means the code starts from the next line
    show_ticks          = yes
    show_tick_labels    = no
    &amp;lt;ticks&amp;gt;
    radius           = 1r
    color            = black
    thickness        = 2p
    multiplier       = 1e-6 
    format           = %d # %d
    &amp;lt;tick&amp;gt;
    spacing        = 1u
    size           = 5p
    &amp;lt;/tick&amp;gt;

    &amp;lt;tick&amp;gt;
    thickness      = 4p
    spacing        = 5u
    size           = 10p
    show_label     = yes
    label_size     = 10p
    label_offset   = 10p
    format         = %d
    &amp;lt;/tick&amp;gt;
    &amp;lt;/ticks&amp;gt; 
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;Creating circos.conf to contain all aprameters and include ticks.conf by adding &amp;quot;&amp;lt;&amp;lt;include ./etc/ticks.conf&amp;gt;&amp;gt;&amp;quot;:&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;    show_scatter   = yes
    show_line      = yes
    show_histogram = yes
    show_heatmaps*   = yes
    show_tile      = yes
    show_highlight = yes
    use_rules      = yes
    #specify input file that contains length of each chromosome of genome
    karyotype = karyotype.acorus.txt
    #specify the order of chromosome
    chromosomes_order = aco1,aco2,aco3,aco4,aco5,aco6,aco7,aco8,aco9,aco10,aco11,aco12
    chromosomes_units = 1000000
    #chromosomes_reverse = /aco/ #reverse
    #chromosomes_scale = /aco/=0.7rn,/anc/=0.3rn
    #chromosomes_display_default = no
    #chromosomes = /aco/;anc1
    #chromosomes_radius = /aco/:0.8r
    chromosomes_color = aco1=rdylgn-11-div-1,aco2=rdylgn-11-div-2,aco3=rdylgn-11-div-3,aco4=rdylgn-11-div-4,aco5=rdylgn-11-div-5,aco6=rdylgn-11-div-6,aco7=rdylgn-11-div-7,aco8=rdylgn-11-div-8,aco9=rdylgn-11-div-9,aco10=rdylgn-11-div-10,aco11=rdylgn-11-div-11,aco12=[#00441b]

    &amp;lt;&amp;lt;include ./etc/ticks.conf&amp;gt;&amp;gt;

    &amp;lt;ideogram&amp;gt;
    &amp;lt;spacing&amp;gt;
    default = 1u
    &amp;lt;/spacing&amp;gt;

    radius           = 0.90r
    thickness        = 20p
    fill             = yes
    stroke_color     = dgrey
    stroke_thickness = 2p

    show_label     = yes #show label
    label_font     = default #font
    label_radius   = dims(ideogram,radius) + 0.08r #location
    label_size     = 16 # font size
    label_parallel = yes 

    label_format   = eval(sprintf(&amp;quot;%s&amp;quot;,var(chr))) # format
    &amp;lt;/ideogram&amp;gt;
    &amp;lt;&amp;lt;include ./etc/links.conf&amp;gt;&amp;gt;
    &amp;lt;image&amp;gt;
    dir*    = .    # output dir
    file*	= Acorus.circos.png
    radius* = 400p # ridus of pic
    svg*    = no   # whether generating svg file
    #angle_offset* = -55
    &amp;lt;&amp;lt;include etc/image.conf&amp;gt;&amp;gt;
    &amp;lt;/image&amp;gt;

    &amp;lt;plots&amp;gt;
    &amp;lt;plot&amp;gt;
    type    = heatmap
    file    = genes_num.txt
    color   = blues-9-seq
    r1      = 0.70r
    r0      = 0.61r
    &amp;lt;/plot&amp;gt;

    &amp;lt;plot&amp;gt;
    type = scatter
    fill_color       = black 
    stroke_color     = black
    glyph            = circle
    glyph_size       = 5 
    file = genes_num.txt
    r1   = 0.80r
    r0   = 0.71r
    &amp;lt;/plot&amp;gt;

    &amp;lt;plot&amp;gt;
    type = histogram
    file = genes_num.txt
    fill_color = blue 
    r1   = 0.89r
    r0   = 0.81r
    &amp;lt;/plot&amp;gt;
    &amp;lt;/plots&amp;gt;

    &amp;lt;&amp;lt;include etc/colors_fonts_patterns.conf&amp;gt;&amp;gt;
    &amp;lt;&amp;lt;include etc/housekeeping.conf&amp;gt;&amp;gt;
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;iv): Preparing input file &amp;quot;genes_num.txt&amp;quot; for heatmap, scatter, line and histpgram&lt;br&gt;
circos requires the format of input format is as &lt;code&gt;chr start end value [options]&lt;/code&gt; which is 4 col of BED  file in fact.&lt;br&gt;
First of all, we need to install bedtools  by &lt;code&gt;conda install -c bioconda bedtools&lt;/code&gt;. (if path is not included in mac than we need to add it)&lt;br&gt;
I just choose first 3 cols (chr, strat, end) of genome&#39;s BED after using jcvi as &amp;quot;genes.bed&amp;quot; and be careful that the first col of genes.bed is coresponding to the ID col of karyotype.genomeA.txt file. (if this is not coreponding, then we will get number zero on the last col of &amp;quot;genes_num.txt&amp;quot; file) For example, in genes.bed which should look like:  &lt;code&gt;aco1	1526	2561&lt;/code&gt; . After that we use bedtool to create windows with setting 500kb by the follwing command:&lt;br&gt;
&lt;code&gt;cut -d &#39; &#39; -f 3,6 karyotype.genomeA.txt | tr &#39; &#39; &#39;\t&#39; &amp;gt; genomeA.genome&lt;/code&gt;&lt;br&gt;
&lt;code&gt;bedtools makewindows -g genomeA.genome -w 500000 &amp;gt; genomeA.windows&lt;/code&gt;&lt;br&gt;
Last step to collect info:&lt;br&gt;
&lt;code&gt;bedtools coverage -a genomeA.windows -b genes.bed | cut -f 1-4 &amp;gt; genes_num.txt&lt;/code&gt;&lt;br&gt;
Therefor, genes_num.txt is our input file.&lt;/p&gt;
&lt;p&gt;v): Eventually, we run the implement the command:&lt;br&gt;
&lt;code&gt;circos -conf circos.conf&lt;/code&gt;&lt;br&gt;
we will get a picture with multiple plots for genome A&lt;br&gt;
reference link :&lt;br&gt;
https://www.jianshu.com/p/ea3a8933ace9&lt;/p&gt;
&lt;h2 id=&#34;plotting-synteny-blocks-between-two-different-genomes&#34;&gt;Plotting synteny blocks between two different genomes&lt;/h2&gt;
&lt;p&gt;Except for all files above, we need to create a link.conf under etc folder to contain all parameters for plotting and a link txt file including all syntenys under circos folder.&lt;br&gt;
i): the format of link txt file is at least 6 cols as &lt;code&gt;chr1 start1 end1 chr2 start2 end2 [options]&lt;/code&gt;&lt;br&gt;
Here we need to use JCVI to get &lt;code&gt;anc.aco.anchors.simple&lt;/code&gt; file and than we have to replace the genename of anchor file as the start and end loction info which is the links input format of circos. For that, we use &lt;code&gt;simple2links.py&lt;/code&gt; from GengzhouXu&#39;s github https://github.com/xuzhougeng/myscripts&lt;br&gt;
&lt;code&gt;simple2links.py anc.aco.anchors.simple&lt;/code&gt;&lt;br&gt;
So we get an output files called &lt;code&gt;anc.aco.anchors.simple_link.txt&lt;/code&gt;&lt;br&gt;
But we need to chnage the chr names of the two genomes anc and aco as &amp;quot;anc&amp;quot; + &amp;quot;chr&amp;quot;, &amp;quot;aco&amp;quot; + &amp;quot;chr&amp;quot; that is coresponding to the &lt;code&gt;karyotype = karyotype.anc.txt,karyotype.aco.txt&lt;/code&gt; from circos.conf&lt;br&gt;
ii): links.conf&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;    &amp;lt;links&amp;gt;
    &amp;lt;link&amp;gt;
    file          = ancestor1.acorus.anchors.simple_link.txt
    radius        = 0.65r
    color         = blue_a4
    ribbon = yes
    bezier_radius = 0r
    &amp;lt;rules&amp;gt;
    &amp;lt;rule&amp;gt;#rule here is setting different colors for links on conditions
    condition = var(chr1) eq &amp;quot;anc1&amp;quot;
    color=rdylgn-7-div-1
    &amp;lt;/rule&amp;gt;
    &amp;lt;rule&amp;gt;
    condition = var(chr1) eq &amp;quot;anc2&amp;quot;
    color=rdylgn-7-div-2
    &amp;lt;/rule&amp;gt;
    &amp;lt;rule&amp;gt;
    condition = var(chr1) eq &amp;quot;anc3&amp;quot;
    color=rdylgn-7-div-3
    &amp;lt;/rule&amp;gt;
    &amp;lt;rule&amp;gt;
    condition = var(chr1) eq &amp;quot;anc4&amp;quot;
    color=165,105,189
    &amp;lt;/rule&amp;gt;
    &amp;lt;rule&amp;gt;
    condition = var(chr1) eq &amp;quot;anc5&amp;quot;
    color=rdylgn-7-div-5
    &amp;lt;/rule&amp;gt;
    &amp;lt;rule&amp;gt;
    condition = var(chr1) eq &amp;quot;anc6&amp;quot;
    color=rdylgn-7-div-6
    &amp;lt;/rule&amp;gt;
    &amp;lt;rule&amp;gt;
    condition = var(chr1) eq &amp;quot;anc7&amp;quot;
    color=rdylgn-7-div-7
    &amp;lt;/rule&amp;gt;
    &amp;lt;/rules&amp;gt;

    &amp;lt;/link&amp;gt;
    &amp;lt;/links&amp;gt;
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;iii): create a new &lt;code&gt;circos.conf&lt;/code&gt; in the current circos folder as:&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;    karyotype = karyotype.ancestor1_resize.txt,karyotype.acorus.txt
    chromosomes_order = aco12,aco11,aco10,aco1,aco4,aco9,aco8,aco7,aco2,aco6,aco3,aco5
    chromosomes_units = 1000000# this is the unit of chr, 1000000 = 1000000kb = 1u
    chromosomes_reverse = /aco/
    chromosomes_scale = /aco/=0.7rn,/anc/=0.3rn
    #chromosomes_display_default = no
    #chromosomes = /aco/;anc1
    chromosomes_radius = /anc/:0.8r
    &amp;lt;&amp;lt;include ./etc/ticks.conf&amp;gt;&amp;gt; #####this command inlude the ticks.conf under etc folder
    &amp;lt;ideogram&amp;gt;
    &amp;lt;spacing&amp;gt;
    default = 1u
    #to get the two genomes symmetric outlet, aco1 face anc1, aco12 face anc7
    &amp;lt;pairwise aco1 anc1&amp;gt;
    spacing = 10u
    &amp;lt;/pairwise&amp;gt;
    
    &amp;lt;pairwise aco12 anc7&amp;gt;
    spacing = 10u
    &amp;lt;/pairwise&amp;gt;
    &amp;lt;/spacing&amp;gt;

    radius           = 0.90r
    thickness        = 20p
    fill             = yes
    stroke_color     = dgrey
    stroke_thickness = 2p

    show_label     = yes #show label
    label_font     = default 
    label_radius   = dims(ideogram,radius) + 0.08r #
    label_size     = 16 # 
    label_parallel = yes # 

    label_format   = eval(sprintf(&amp;quot;%s&amp;quot;,var(chr))) # 
    &amp;lt;/ideogram&amp;gt;

    &amp;lt;&amp;lt;include ./etc/links.conf&amp;gt;&amp;gt; ##### this ocmmand is using links.conf under folder of etc

    &amp;lt;image&amp;gt;
    dir*    = .    # output dir, * means we can update this value of the variable
    radius* = 400p # 
    svg*    = no   # 
    angle_offset* = -55
    &amp;lt;&amp;lt;include etc/image.conf&amp;gt;&amp;gt;
    &amp;lt;/image&amp;gt;

    &amp;lt;&amp;lt;include etc/colors_fonts_patterns.conf&amp;gt;&amp;gt;
    &amp;lt;&amp;lt;include etc/housekeeping.conf&amp;gt;&amp;gt;
&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;The last step run command &lt;code&gt;circos -conf circos.conf&lt;/code&gt;&lt;/p&gt;
&lt;p&gt;&lt;strong&gt;Reference link&lt;/strong&gt;:https://www.jianshu.com/p/1658e702ba17.&lt;br&gt;
We can split chr as several part by : http://www.360doc.com/content/19/1224/13/68068867_881784578.shtml .&lt;br&gt;
Or check my folder ~/circos/Acorus_anc1/newchr1_aco&lt;/p&gt;
">How to use CIRCOS plot pairwise syteny blocks</a>
      </div>
      
    </div>
    <div class="page">
      <div id="page_ul"></div>
    </div>
  </div>
</div>
<script>
  !function () {
    let searchMask = document.querySelector('#search_mask');
    let result = document.querySelector('#result');
    let items = document.querySelectorAll('.item');
    let searchBox = document.querySelector('#search');
    let statCount = document.querySelector('#stat_count');
    let statTimes = document.querySelector('#stat_times');
    let pageUl = document.querySelector('#page_ul');
    let close = document.querySelector('#close');
    
    close.addEventListener('click', function() {
      searchMask.style = 'display: none;'
    })

    let finds = [];
    let contents = [];
    let pageSize = 10;
    items.forEach(item => {
      let a = item.querySelector('a');
      contents.push({
        title: a.innerText,
        details: a.dataset.c,
        link: a.href
      })
      item.remove();
    })

    function insertStr(soure, start, count) {
      let newStr = soure.substr(start, count);
      return soure.slice(0, start) + '<em>' + newStr + '</em>' + soure.slice(start + count);
    }

    pageUl.addEventListener('click', function(event) {
      let target = event.target;
      if (target.__proto__ === HTMLSpanElement.prototype) {
        appendResults(parseInt(target.dataset.i));
      }
    })

    function appendResults(index) {
      let htmlResult = '';
      let start = index || 0;
      let end = Math.min(start + pageSize, finds.length);
      for (let i = start; i < end; i++) {
        const current = finds[i];
        let html = current.title;
        let sum = 0;
        let positions = current.positions;
        positions.forEach(position => {
          html = insertStr(html, position.start + sum, position.count);
          sum += 9;
        })
        htmlResult += `<div class="item"><a class="result-title" href="${current.link}">${html}</a></div>`;
      }
      result.innerHTML = htmlResult;
      pageUl.innerHTML = '';
      let count = finds.length / pageSize;
      let lis = '';
      if (start !== 0) {
        lis += `<span class="fa fa-angle-left" data-i='${start - 1}'></span>`;
      }
      for (let i = 0; i < count; i++) {
        lis += `<span class='${i === start?'current':''}' data-i='${i}'>${i+1}</span>`;     
      }
      if (start+1 < count) {
        lis += `<span class="fa fa-angle-right" data-i='${start+1}'></span>`;  
      }
      pageUl.innerHTML = lis;
    }

    function search(delay) {
      let timer = null
      return function () {
        clearTimeout(timer)
        timer = setTimeout(() => {
          let start = Date.now();
          let segments = searchBox.value.split(' ').filter(c => c != '');
          if (segments.length <= 0) {
            return;
          }
          finds = [];
          let htmlResult = '';
          contents.forEach(content => {
            let title = content.title;
            let positions = [];
            let find = false;
            segments.forEach((segment) => {
              if (content.title.includes(segment)) {
                find = true;
                positions.push({
                  start: content.title.indexOf(segment),
                  count: segment.length
                })
              } else if (content.details.includes(segment)) {
                find = true;
              }
            });
            if (find) {
              finds.push({
                title: content.title,
                link: content.link,
                positions
              });
            }
          })
          appendResults(0);
          statCount.textContent = finds.length;
          statTimes.textContent = Date.now() - start;
        }, delay)
      }
    }
    searchBox.addEventListener('input', search(200));
  }()
</script>

<input hidden id="copy" />
<script>
  !function () {
    let times = document.querySelectorAll('.publish-time');
    for (let i = 0; i < times.length; i++) {
      let date = times[i].dataset.t;
      let time = Math.floor((new Date().getTime() - new Date(date).getTime()) / 1000);
      if (time < 60) {
        str = time + 'Áßí‰πãÂâç';
      } else if (time < 3600) {
        str = Math.floor(time / 60) + 'ÂàÜÈíü‰πãÂâç';
      } else if (time >= 3600 && time < 86400) {
        str = Math.floor(time / 3600) + 'Â∞èÊó∂‰πãÂâç';
      } else if (time >= 86400 && time < 259200) {
        str = Math.floor(time / 86400) + 'Â§©‰πãÂâç';
      } else {
        str = times[i].textContent;
      }
      times[i].textContent = str;
    }
  }();
</script>

<script>
  let language = 'en';
  if (language !== '') {
    let map = new Map();
    if (language === 'en') {
      map.set('search', 'Search');
      map.set('category', 'Categories');
      map.set('article', 'Articles');
      map.set('tag', 'Tags');
      map.set('top', 'Top');
      map.set('publish', 'published');
      map.set('minute', ' minutes');
      map.set('read-more', 'Read More');
      map.set('view', 'View');
      map.set('words', ' words');
      map.set('category-in', 'category in');
      map.set('preview', 'Meta');
      map.set('index', 'Toc');
      map.set('no-archives', "You haven't created yet");
      map.set('archives', " articles in total");
      map.set('cloud-tags', " tags in total");
      map.set('copyright', "Copyright: ");
      map.set('author', "Author: ");
      map.set('link', "Link: ");
      map.set('leave-message', "Leave a message");
      map.set('format', "Links Format");
      map.set('site-name', "Name: ");
      map.set('site-link', "Link: ");
      map.set('site-desc', "Desc: ");
      map.set('stat', " related results, taking ");
      map.set('stat-time', " ms");
      map.set('site-img', "Image: ");
    }

    if (map.size > 0) {
      let lanElems = document.querySelectorAll('.language');
      lanElems.forEach(elem => {
        let lan = elem.dataset.lan, text = map.get(lan);
        if (elem.__proto__ === HTMLInputElement.prototype) {
          elem.placeholder = text
        } else {
          if (elem.dataset.count) {
            text = elem.dataset.count + text;
          }
          elem.textContent = text;
        }
      })
    }
  }
  //ÊãøÊù•‰∏ª‰πâ(ÁúüÈ¶ô)^_^ÔºåClipboard ÂÆûÁé∞ÊëòËá™ÊéòÈáë https://juejin.im/post/5aefeb6e6fb9a07aa43c20af
  window.Clipboard = (function (window, document, navigator) {
    var textArea,
      copy;

    // Âà§Êñ≠ÊòØ‰∏çÊòØiosÁ´Ø
    function isOS() {
      return navigator.userAgent.match(/ipad|iphone/i);
    }
    //ÂàõÂª∫ÊñáÊú¨ÂÖÉÁ¥†
    function createTextArea(text) {
      textArea = document.createElement('textArea');
      textArea.value = text;
      textArea.style.width = 0;
      textArea.style.height = 0;
      textArea.clientHeight = 0;
      textArea.clientWidth = 0;
      document.body.appendChild(textArea);
    }
    //ÈÄâÊã©ÂÜÖÂÆπ
    function selectText() {
      var range,
        selection;

      if (isOS()) {
        range = document.createRange();
        range.selectNodeContents(textArea);
        selection = window.getSelection();
        selection.removeAllRanges();
        selection.addRange(range);
        textArea.setSelectionRange(0, 999999);
      } else {
        textArea.select();
      }
    }

    //Â§çÂà∂Âà∞Ââ™Ë¥¥Êùø
    function copyToClipboard() {
      try {
        document.execCommand("Copy")
      } catch (err) {
        alert("Â§çÂà∂ÈîôËØØÔºÅËØ∑ÊâãÂä®Â§çÂà∂ÔºÅ")
      }
      document.body.removeChild(textArea);
    }

    copy = function (text) {
      createTextArea(text);
      selectText();
      copyToClipboard();
    };

    return {
      copy: copy
    };
  })(window, document, navigator);

  function copyCode(e) {
    if (e.srcElement.tagName === 'SPAN' && e.srcElement.classList.contains('copy-code')) {
      let code = e.currentTarget.querySelector('code');
      var text = code.innerText;
      if (e.srcElement.textContent === 'Â§çÂà∂ÊàêÂäü') {
        return;
      }
      e.srcElement.textContent = 'Â§çÂà∂ÊàêÂäü';
      (function (elem) {
        setTimeout(() => {
          if (elem.textContent === 'Â§çÂà∂ÊàêÂäü') {
            elem.textContent = 'Â§çÂà∂‰ª£Á†Å'
          }
        }, 1000);
      })(e.srcElement)
      Clipboard.copy(text);
    }
  }

  let pres = document.querySelectorAll('pre');
  pres.forEach(pre => {
    let code = pre.querySelector('code');
    let copyElem = document.createElement('span');
    copyElem.classList.add('copy-code');
    copyElem.textContent = 'Â§çÂà∂‰ª£Á†Å';
    pre.appendChild(copyElem);
    pre.onclick = copyCode
  })

</script>
<script src="/media/js/motion.js"></script>

<script src="https://cdn.jsdelivr.net/gh/cferdinandi/smooth-scroll/dist/smooth-scroll.polyfills.min.js"></script>
<script>
  var scroll = new SmoothScroll('a[href*="#"]', {
    speed: 200
  });
</script>





  <div class="snow-container"></div>
  <script src="/media/js/bg/canvas-nest.js"></script>
  <script>
  const config = {
    color: '255,0,0',
    count: 88,
  };
  // Using config rendering effect at 'element'.
  const cn = new CanvasNest(document.querySelector('.snow-container'), config);
  </script>


</html>